src/lib.rs

```diff
@@ -1,4 +1,36 @@
+use async_trait::async_trait;
+use std::io;
+
 pub mod fluent;
-pub mod gecko;
 pub mod registry;
 pub mod source;
+pub mod tokio;
+
+mod futures;
+
+/// Naming things is hard. The consumers of `fluent-fallback` implement this
+/// trait to provide loading of fluent resources, returning a `String`
+/// representating the resource.
+///
+/// Possible alternate names: FileSource, ResourceSource, StringLoader.
+///
+/// TODO: This should live in a fluent crate. But circular dependencies.
+#[async_trait]
+pub trait StringFetcher {
+    /// Return the `String` representation for `resource_id`. This version is
+    /// blocking. See `fetch_async`.
+    ///
+    /// Note: Possible alternate name: `read_sync`, `load_async`.  `fetch` works
+    /// is a generic way, meaning `get me the string for, I don't care how`.
```

Comment:
I like `fetch` and `fetch_sync` for the function that doesn't do logic, just fetches. I use `load` for cases which may cache, or perform some cache invalidation.